MapboxNavigation/RouteTableViewController.swift

```diff
@@ -44,8 +45,45 @@ class RouteTableViewController: UIViewController {
         
         if routeProgress.durationRemaining < 60 {
             headerView.timeRemaining.text = String.localizedStringWithFormat(NSLocalizedString("LESS_THAN", bundle: .mapboxNavigation, value: "<%@", comment: "Format string for less than; 1 = duration remaining"), dateComponentsFormatter.string(from: 61)!)
+            headerView.timeRemaining.textColor = TimeRemainingLabel.appearance(for: traitCollection).textColor
         } else {
             headerView.timeRemaining.text = dateComponentsFormatter.string(from: routeProgress.durationRemaining)
+            
+            let coordinatesRemainingOnStep = Int(floor((Double(routeProgress.currentLegProgress.currentStepProgress.step.coordinateCount)) * routeProgress.currentLegProgress.currentStepProgress.fractionTraveled))
+            
+            guard coordinatesRemainingOnStep >= 0 else {
+                headerView.timeRemaining.textColor = TimeRemainingLabel.appearance(for: traitCollection).textColor
+                return
+            }
+            
+            let segementsForStep = routeProgress.congestionTravelTimesSegmentsByStep[routeProgress.legIndex][routeProgress.currentLegProgress.stepIndex]
+            
+            guard coordinatesRemainingOnStep <= segementsForStep.count else { return }
+            
+            let currentStepCongestionTimes = segementsForStep.suffix(from: coordinatesRemainingOnStep)
+            let remainingStepSegments = Array(routeProgress.congestionTravelTimesSegmentsByStep[routeProgress.legIndex].suffix(from: routeProgress.currentLegProgress.stepIndex)).joined()
```

Comment:
If I’m not mistaken, this could still be a huge array unless we find a way to flatten or precompute the expected travel time buckets for later steps. Can we add another dictionary to `RouteProgress` that would make it possible to change this code to something like the following sketch?

```swift
let timedCongestionLevelsLeftOnStep: [TimedCongestionLevel] = stepTimedCongestionLevels.suffix(from: countOfCoordinatesLeftOnStep)
// var travelTimesByCongestionLevelByStep: [[[CongestionLevel: TimeInterval]]]
let timesLeftOnLegByCongestionLevel: [[CongestionLevel: TimeInterval]] = routeProgress.travelTimesByCongestionLevelByStep[routeProgress.legIndex].suffix(from: legProgress.stepIndex)

// `sum(_:)` is really a call to `reduce(_:_:)`
// The for loop before is OK too – just wrote it this way for clarity.
let lowTimeLeftOnStep: TimeInterval = sum(timedCongestionLevelsLeftOnStep.filter { $0.0.low }.map { $0.1 })
let lowTimeLeftOnLeg: TimeInterval = sum(timesLeftOnLegByCongestionLevel.map { $0.low })
let remainingLowTime = lowTimeLeftOnStep + lowTimeLeftOnLeg
```